Purchases/Public/RCPurchases.m

```diff
@@ -223,7 +223,12 @@ - (instancetype)initWithAPIKey:(NSString *)APIKey
                   observerMode:(BOOL)observerMode
                 platformFlavor:(nullable NSString *)platformFlavor
          platformFlavorVersion:(nullable NSString *)platformFlavorVersion {
-    RCReceiptFetcher *receiptFetcher = [[RCReceiptFetcher alloc] init];
+    RCOperationDispatcher *operationDispatcher = [[RCOperationDispatcher alloc] init];
+    RCReceiptRefreshRequestFactory *receiptRefreshRequestFactory = [[RCReceiptRefreshRequestFactory alloc] init];
+    RCStoreKitRequestFetcher *fetcher = [[RCStoreKitRequestFetcher alloc]
+                                         initWithRequestFactory:receiptRefreshRequestFactory
+                                         operationDispatcher:operationDispatcher];
+    RCReceiptFetcher *receiptFetcher = [[RCReceiptFetcher alloc] initWithRequestFetcher:fetcher];
```

Comment:
had to move some stuff around, but the only new bit here is the `receiptFetcher` having the `requestFetcher` injected. 
Previously, `RCPurchases` was just using the `requestFetcher` directly.